src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -1865,6 +1865,40 @@ JNI_METHOD(void, read)
     }
 }
 
+// Convert Json to Tlv, and remove the outer structure
+CHIP_ERROR ConvertJsonToTlvWithoutStruct(const std::string & json, MutableByteSpan & data, size_t & writtenLength)
+{
+    uint8_t buf[chip::app::kMaxSecureSduLengthBytes] = { 0 };
+    TLV::TLVReader tlvReader;
+    TLV::TLVType outerContainer = TLV::kTLVType_Structure;
+    MutableByteSpan dataWithStruct{ buf };
+    ReturnErrorOnFailure(JsonToTlv(json, dataWithStruct));
+    tlvReader.Init(dataWithStruct);
+    ReturnErrorOnFailure(tlvReader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+    ReturnErrorOnFailure(tlvReader.EnterContainer(outerContainer));
+    ReturnErrorOnFailure(tlvReader.Next());
+
+    TLV::TLVWriter tlvWrite;
+    tlvWrite.Init(data);
+    ReturnErrorOnFailure(tlvWrite.CopyElement(TLV::AnonymousTag(), tlvReader));
+    ReturnErrorOnFailure(tlvWrite.Finalize());
+    writtenLength = tlvWrite.GetLengthWritten();
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PutPreencodedWriteAttribute(app::WriteClient & writeClient, uint32_t endpointId, uint32_t clusterId,
```

Comment:
How about passing in a `ConcreteDataAttributePath` directly to this method to reduce argument count and not need as many casts in the Put below?